### PR TITLE
Sender chat colors

### PR DESCRIPTION
**Pull request by ccrabb**
The only real ReactJS decision I feel like I made here was to add `senderColor` as a parameter to the `Message` component. Originally I applied the color via the ReactJS `color` property, but then it made the TSX markup a bit unwieldy. All feedback welcome!

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -35,6 +36,11 @@ import useTinyCount from '../useTinyCount'
 import { update } from 'lodash'
 
 const INTRO_SEEN_KEY = 'intro_seen'
+const adminUsers: string[] = [
+  '@wokebot:woke.net',
+  '@chromakode:woke.net',
+  '@tree:woke.net'
+]
```
Comment (chromakode) [inline src/pages/index.tsx]: Let's hold off from differentiating admins right now and split this into a separate project / discussion. I'm not sure if/how we'd like to display this, and the status should be sourced from the matrix user "power level" (>=50 means the user is a mod, 100 means the user is an admin)

File: src/pages/index.tsx [hunk 2]
```diff
@@ -71,6 +77,42 @@ function Home() {
...
+
+  function idColor(id: string) {
+    if (!id) {
+      return Color('white')
+    }
+    const h = hashText(id, 60)
+    const sPart = hashText(id, 60)
+    return Color({ h, s: 70 + sPart, l: 60 })
+  }
+
```
Comment (chromakode) [inline src/pages/index.tsx]: Could you please move these functions to a separate file -- perhaps `src/colors.ts`?

File: src/pages/index.tsx [hunk 3]
```diff
@@ -207,13 +249,16 @@ function Home() {
...
                 }
+
                 return (
                   <Message
                     key={ev.event.event_id}
                     sender={ev.sender.name}
+                    senderColor={getSenderColor(ev.sender.userId)}
                     body={ev.event.content.body}
                     px={4}
-                    _odd={{ backgroundColor: 'gray.900' }}
```
Comment (chromakode) [inline src/pages/index.tsx:216]: I'd like to retain the even/odd background coloring for now.

File: src/pages/index.tsx [hunk 4]
```diff
@@ -248,8 +293,10 @@ function Home() {
                 focusBorderColor="flame.600"
                 placeholder="Say something"
                 value={messageText}
+                maxLength={500}
                 onChange={(ev) => setMessageText(ev.target.value)}
               />
+              {messageText.length > 250 ? <small>{messageText.length}/500</small> : null}
```
Comment (chromakode) [inline src/pages/index.tsx]: Ooo This is awesome. I'd like to see how this looks -- I've tweaked our Netlify settings to the next time you push it'll build a site preview for us to check out. In the future, could you please split out separate features like this into different PRs? Will make it faster to review/merge things piecemeal.

**Review by chromakode — CHANGES_REQUESTED** [review]
Great work! A couple small requests before we ship this.

File: src/pages/index.tsx (hunk 4 above)
Comment (chromakode) [reply src/pages/index.tsx]: One more thought: the 500 char limit is a pretty important global value. Could you please move it to https://github.com/wokenet/anchor/blob/main/constants.json?

**Review by chromakode — CHANGES_REQUESTED** [review]

File: src/pages/index.tsx (hunk 4 above)
Comment (ccrabb) [reply src/pages/index.tsx]: Yep to both!

File: src/pages/index.tsx (hunk 3 above)
Comment (ccrabb) [reply src/pages/index.tsx:216]: Okay

File: src/pages/index.tsx [hunk 5]
```diff
@@ -241,15 +247,21 @@ function Home() {
...
+                  flex={1}
+                  focusBorderColor="flame.600"
+                  placeholder="Say something"
+                  value={messageText}
+                  maxLength={maximumMessageSize}
+                  onChange={(ev) => setMessageText(ev.target.value)}
+                />
+                <InputRightElement flexShrink={0} width={50} height={50}>
+                  {messageText.length > 250 ? <small>{maximumMessageSize-messageText.length}</small> : null}
+                </InputRightElement>
```
Comment (chromakode) [inline src/pages/index.tsx]: Could we please hide this when no limit is near, so it doesn't constrain the available text space?

```suggestion
                {messageText.length > 250 && (
                  <InputRightElement flexShrink={0} width={50} height={50}>
                    <small>{maximumMessageSize-messageText.length}</small>
                  </InputRightElement>
                )}
```

File: src/pages/index.tsx [hunk 6]
```diff
@@ -241,15 +247,21 @@ function Home() {
...
+                  m={2}
+                  px={2}
+                  flex={1}
+                  focusBorderColor="flame.600"
+                  placeholder="Say something"
+                  value={messageText}
+                  maxLength={maximumMessageSize}
+                  onChange={(ev) => setMessageText(ev.target.value)}
+                />
+                <InputRightElement flexShrink={0} width={50} height={50}>
```
Comment (chromakode) [inline src/pages/index.tsx]: We should make the limit text unselectable with:

```suggestion
                <InputRightElement flexShrink={0} width={50} height={50} pointerEvents="none">
```

**Review by chromakode — APPROVED** [review]

**Pull request merged**